django/contrib/staticfiles/storage.py

```diff
@@ -274,7 +282,56 @@ def path_level(name):
 
                 # and then set the cache accordingly
                 hashed_files[self.hash_key(name)] = hashed_name
+
+                yield name, hashed_name, processed, substitutions
+
+    def post_process(self, paths, dry_run=False, **options):
+        """
+        Post process the given OrderedDict of files (called from collectstatic).
+
+        Processing is actually two separate operations:
+
+        1. renaming files to include a hash of their content for cache-busting,
+           and copying those files to the target storage.
+        2. adjusting files which contain references to other files so they
+           refer to the cache-busting filenames.
+
+        If either of these are performed on a file, then that file is considered
+        post-processed.
+        """
+        # don't even dare to process the files if we're in dry run mode
+        if dry_run:
+            return
+
+        # where to store the new paths
+        hashed_files = OrderedDict()
+
+        # build a list of adjustable files
+        adjustable_paths = [
+            path for path in paths
+            if matches_patterns(path, self._patterns.keys())
+        ]
+
+        # Do a single pass first, this will post-process all files once, then we can repeat for adjustable files
+        for name, hashed_name, processed, _ in self._post_process(paths, adjustable_paths, hashed_files):
+            yield name, hashed_name, processed
+
+        paths = {
+            path: paths[path] for path in adjustable_paths
+        }
+
+        for i in range(0, self.max_post_process_passes):
+            substitutions = False
+
+            for name, hashed_name, processed, subst in self._post_process(paths, adjustable_paths, hashed_files):
                 yield name, hashed_name, processed
+                substitutions = substitutions or subst
+
+            if not substitutions:
+                break
+
+        if substitutions:
+            yield 'All', None, RuntimeError("Max post-process passes exceeded")
```

Comment:
Dunno if `RuntimeError` is the best exception to use here, will only come up when the user is running `collectstatic`